### ISSUE TITLE
Pressing enter when mid-line in an existing comment should auto comment the rest of the line when it moves down.

### ISSUE THREAD

**Issue by asdasd-123** [kbrose/vsc-python-indent#5]
Easier to probably demonstrate:

Lets say we have the following line (pretend the cursor is the £ symbol):
`# test test £test test`

when you press enter, it should become:
```
# test test 
# test test
```

instead, it becomes:
```
# test test
test test
```
(missing the hash+space)

**Comment by kbrose** [comment]
Interesting idea! I'd be happy to review a PR that adds this functionality.

**Comment by asdasd-123** [comment]
I have no idea how to add it sorry. My point was that the IDE does this by default but this plugin breaks it lol

**Comment by kbrose** [comment]
I'd written less than one hundred lines of javascript/typescript before writing this extension. Don't sell yourself short!

**Comment by WhistleWhileYouWork** [comment]
Just wanted to put my 2 cents in. I do this a lot! I constantly have to go back and resize my comments after I end up indenting a block and shortening the horizontal space. I'll have another look at the code but it seems the section that looks for the `#` just exits after it is detected. Is this where new code needs to go?

1. Detect if there is text after the cursor
2. Make new line indent same as current line
3. Add `# ` after the indent and before the cursor with remaining text after cursor

With this issue fixed, this extension should be complete!

**Comment by kbrose** [comment]
I would probably make this a separate function instead of trying to add another output to the already very complex parser. 

Something else to think about is the behavior if there is non-comment text. How should the below work?

```python
def f(x):  # this is| gonna be great!
```

(The pipe represents your cursor.)

I’d make a simple function that checks the line is only white space and comment, and that your cursor is inside the comment, and in that case sets the insertion text to include `# `

**Comment by WhistleWhileYouWork** [comment]
That was the only edge case I could come up with as well, and I believe the appropriate action is to follow VSCode's behavior which is to not comment the following line. This makes sense as there should be no expectation by the programmer to have the comment continue on the next line since single line code/comments are meant to be just that - a single line. I have never seen a same-line code comment continue to the line below.

This will take some time for me to get to, but I would like to see this fixed since this one is frustrating (and as far as I'm concerned the final bug).

**Comment by kbrose** [comment]
Should be working in v1.5.0. Please reopen/open a new issue if the problem persists.

I matched the behavior of vanilla VSCode, even though I think it's sub-optimal in the case when you have a space to the right of your cursor (you end up with two spaces):

```python
def f():
    # This function is| gonna be REAL good!

def f():
    # This function is
    # | gonna be REAL good
```

PRs for better behavior welcome!

**Comment by WhistleWhileYouWork** [comment]
Nice!

I don't think it's sub-optimal at all. The result is exactly what I would expect. Because I know I'm entering a new line, I always highlight the space before hitting ENTER. I don't think the extension should try to be too 'smart'. Just look at the issues with the latest version of the python extension: https://github.com/microsoft/vscode-python/issues/6886

**Comment by kbrose** [comment]
Yeah, that behavior was driving me nuts today. I thought I went crazy and spent a good amount of time trying to figure out how to turn it off. I discovered it while pasting and was really confused why I was getting format on paste when I specifically turned ***off*** the format-on-paste setting.

I do think a little intelligence wouldn't be terrible here. If there is (possibly exactly) one space to the right of the cursor, do not add a space, just a `#`.But I obviously didn't feel strong enough to do it myself, so /shrug
